nestkernel/connector_base.h

```diff
@@ -75,6 +75,37 @@ suicide_and_resurrect( Told* connector, C connection )
   return p;
 }
 
+template < typename Tnew, typename Told >
+inline Tnew*
+suicide_and_resurrect( Told* connector, size_t i )
+{
+#ifdef USE_PMA
+#ifdef IS_K
+  Tnew* p =
+    new ( poormansallocpool[ omp_get_thread_num() ].alloc( sizeof( Tnew ) ) ) Tnew( *connector, i );
+#else
+  Tnew* p = new ( poormansallocpool.alloc( sizeof( Tnew ) ) ) Tnew( *connector, i );
+#endif
+  connector->~Told();
+#else
+  Tnew* p = new Tnew( *connector, i );
+  delete connector; // suicide
+#endif
+  return p;
+}
+
+template < typename Tnew, typename Told >
+inline Tnew*
+suicide( Told* connector )
+{
+  Tnew* p = 0;
+#ifdef USE_PMA
+  connector->~Told();
+#else
+  delete connector; // suicide
+#endif
+  return p;
```

Comment:
Why does `suicide` have to return `Tnew*` if it will return 0 any way? It should be a function returning `void`.